src/wallet/rpcdump.cpp

```diff
@@ -575,15 +580,34 @@ UniValue importwallet(const JSONRPCRequest& request)
         int64_t nFilesize = std::max((int64_t)1, (int64_t)file.tellg());
         file.seekg(0, file.beg);
 
+        static const std::string HEADER_STRING = "# Wallet dump created by Bitcoin";
+        if (file.good()) {
+            std::string line;
+            std::getline(file, line);
+
+            // Check that the first line has HEADER_STRING as a prefix.
```

Comment:
Yes, besides the 2 warnings added in this commit the comment lines are ignored. My reasoning for checking them is that it might give a useful hint to the user if the result of using this RPC isn't what they intended (e.g. fewer keys/scripts imported than intended).  Maybe it's not as useful with the log statement that gives the number of key/scripts imported and skipped. 

I would  say it's more of a bug fix than a breaking change. It seems correct that `importwallet /dev/null`, or accidentally importing a text file should report an error. Unless you think the error condition (nothing skipped/imported) is too strict to be considered a bug fix.

Reply:
You can still check and report everything beside comments or am I missing something?

Reply:
No, that's right.